app/scripts/modules/core/src/task/monitor/TaskMonitor.ts

```diff
@@ -31,14 +32,17 @@ export class TaskMonitor {
   public modalInstance: IModalServiceInstance;
   private monitorInterval: number;
   private onTaskComplete: () => any;
-  private onTaskRetry: () => void;
+  public onTaskRetry: () => void;
+  public statusUpdatedStream: Subject<void> = new Subject<void>();
```

Comment:
Hi, this is React, and we just mutate the heck out of the TaskMonitor object as it receives updates, so we don't get a re-render without some hook to indicate the thing is mutated.

(It gets worse, see next comment)